service/storage_proxy.hh

```diff
@@ -235,6 +235,8 @@ public:
     // Get information about this node's view update backlog. It combines information from all local shards.
     db::view::update_backlog get_view_update_backlog();
 
+    future<std::optional<db::view::update_backlog>> get_view_update_backlog_if_changed(std::optional<db::view::update_backlog> last_backlog);
```

Comment:
Nit: you require that this function must be called on shard 0 only. Please put this information in a comment near the signature so that the future users are not surprised.

Reply:
Added